src/main/java/frc/robot/subsystems/shooter/ShooterSubsystem.java

```diff
@@ -0,0 +1,138 @@
+package frc.robot.subsystems.shooter;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.PortConstants;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkPIDController;
+
+// Shoutout to Avni whose code I reverse engineered - Fox in a box(awesome sause)
+
+ public class ShooterSubsystem extends SubsystemBase{
+    //make the motor variables
+    private final CANSparkMax followerMotor;
+    private final CANSparkMax leaderMotor;
+    private final CANSparkMax pivotMotor;
+    private final CANSparkMax flapMotor;
+    //make the motor encoder variables
+    private final RelativeEncoder followerEncoder; //Don't know what to do with this, but its there
+    private final RelativeEncoder leaderEncoder;
+    private final RelativeEncoder pivotEncoder;
+    private final RelativeEncoder flapEncoder;
+    //make the limit switch variables
+    private final DigitalInput pivotLimitSwitch;
+    private final DigitalInput flapLimitSwitch;
+    //PID stuff
+    private static SparkPIDController pivotPidController;
+    private static SparkPIDController flapPidController;
+    private static SparkPIDController leaderPidController;
+    
+
+
+    public ShooterSubsystem(){
+        // intialize motors
+        followerMotor = new CANSparkMax(PortConstants.SHOOTER_FOLLOWER_MOTOR_ID, MotorType.kBrushless);
+        leaderMotor = new CANSparkMax(PortConstants.SHOOTER_LEADER_MOTOR_ID, MotorType.kBrushless);
```

Comment:
you should name the constant in the Constants file TOP/BOTTOM so it's a bit easier to understand when wiring. the names for your motor objects are good though